src/main/python/inverse_parser/_node_restorer.py

```diff
@@ -0,0 +1,684 @@
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.names = _NodeRestorer.restore_many(xml_node_children)
+
+    @staticmethod
+    def restore_ast_import_from(xml_node: ET.Element, py_node: ast.AST) -> None:
+        py_node.module = xml_node.attrib['value'] if 'value' in xml_node.attrib else None
+        py_node.level = int(xml_node.attrib['import_level'])
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.names = _NodeRestorer.restore_many(xml_node_children)
+
+    @staticmethod
+    def restore_ast_alias(xml_node: ET.Element, py_node: ast.AST) -> None:
+        py_node.name = xml_node.attrib['value']
+        xml_identifier = xml_node.find('identifier')
+        py_node.asname = xml_identifier.attrib['value'] if xml_identifier is not None else None
+
+    # - Control flow:
+
+    @staticmethod
+    def restore_ast_if(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.test = _NodeRestorer.restore(xml_node_children[0])
+
+        xml_node_body = _XmlNodeChildrenGetter.get_unique_child(xml_node, with_tag='body')
+        xml_node_body_children = _XmlNodeChildrenGetter.get_children(xml_node_body)
+        py_node.body = _NodeRestorer.restore_many(xml_node_body_children)
+
+        _AttributeSetter.set_list_field(xml_node, py_node, 'orelse')
+
+    @staticmethod
+    def restore_ast_for(xml_node: ET.Element, py_node: ast.AST) -> None:  # TODO: note in README that "type comment"s are not supported by parser
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.target = _NodeRestorer.restore(xml_node_children[0])
+        py_node.iter = _NodeRestorer.restore(xml_node_children[1])
+
+        xml_node_body = _XmlNodeChildrenGetter.get_unique_child(xml_node, with_tag='body')
+        xml_node_body_children = _XmlNodeChildrenGetter.get_children(xml_node_body)
+        py_node.body = _NodeRestorer.restore_many(xml_node_body_children)
+
+        _AttributeSetter.set_list_field(xml_node, py_node, 'orelse')
+
+    @staticmethod
+    def restore_ast_while(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.test = _NodeRestorer.restore(xml_node_children[0])
+
+        xml_node_body = _XmlNodeChildrenGetter.get_unique_child(xml_node, with_tag='body')
+        xml_node_body_children = _XmlNodeChildrenGetter.get_children(xml_node_body)
+        py_node.body = _NodeRestorer.restore_many(xml_node_body_children)
+
+        _AttributeSetter.set_list_field(xml_node, py_node, 'orelse')
+
+    @staticmethod
+    def restore_ast_break(xml_node: ET.Element, py_node: ast.AST) -> None:
+        pass
+
+    @staticmethod
+    def restore_ast_continue(xml_node: ET.Element, py_node: ast.AST) -> None:
+        pass
```

Comment:
Is it ok? Should we implement it? If it is not, describe why in a comment, please

Reply:
Yes, it's ok. This is the valid implementation